api/src/main/java/jakarta/enterprise/lang/model/AnnotationInfo.java

```diff
@@ -1,32 +1,35 @@
 package jakarta.enterprise.lang.model;
 
 import jakarta.enterprise.lang.model.declarations.ClassInfo;
+
+import java.lang.annotation.Annotation;
 import java.lang.annotation.Repeatable;
 import java.util.Collection;
 
-public interface AnnotationInfo {
+/**
+ * Models an annotation definition, providing access to the {@link AnnotationMemberValue}
+ * instances.
+ *
+ * @param <T>  The annotation type.
+ */
+public interface AnnotationInfo<T extends Annotation> {
```

Comment:
I'm not sure if this needs to be parameterized, but let's discuss that later.

(As for `ClassInfo`, I'd pretty much love to remove the type parameter -- I originally added it to be able to express a query, but we don't allow "injecting" `ClassInfo` anymore and we want `ClassConfig` to _not_ inherit from `ClassInfo`, so that reason no longer applies.)